src/unix/epoll.rs

```diff
@@ -0,0 +1,429 @@
+        const ERROR = EPOLLERR as u32;
+        /// This can be used to detect peer shutdown when using Edge Triggered monitoring.
+        const READ_HANG_UP = EPOLLRDHUP as u32;
+        /// Sets the Edge Triggered behavior for the associated file descriptor.
+        /// The default behavior is Level Triggered.
+        const EDGE_TRIGGERED = EPOLLET as u32;
+        /// Hang up happened on the associated file descriptor. Note that `epoll_wait`
+        /// will always wait for this event and it is not necessary to set it in events.
+        const HANG_UP = EPOLLHUP as u32;
+        /// There is an exceptional condition on that file descriptor. It is mostly used to
+        /// set high priority for some data.
+        const PRIORITY = EPOLLPRI as u32;
+        /// The event is considered as being "processed" from the time when it is returned
+        /// by a call to `epoll_wait` until the next call to `epoll_wait` on the same
+        /// epoll file descriptor, the closure of that file descriptor, the removal of the
+        /// event file descriptor via EPOLL_CTL_DEL, or the clearing of EPOLLWAKEUP
+        /// for the event file descriptor via EPOLL_CTL_MOD.
+        const WAKE_UP = EPOLLWAKEUP as u32;
+        /// Sets the one-shot behavior for the associated file descriptor.
+        const ONE_SHOT = EPOLLONESHOT as u32;
+        /// Sets an exclusive wake up mode for the epoll file descriptor that is being
+        /// attached to the associated file descriptor.
+        /// When a wake up event occurs and multiple epoll file descriptors are attached to
+        /// the same target file using this mode, one or more of the epoll file descriptors
+        /// will receive an event with `epoll_wait`. The default here is for all those file
+        /// descriptors to receive an event.
+        const EXCLUSIVE = EPOLLEXCLUSIVE as u32;
+    }
+}
+
+/// Wrapper over 'libc::epoll_event'.
+///
+// We are using `transparent` here to be super sure that this struct and its fields
+// have the same alignment as those from the `epoll_event` struct from C.
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct EpollEvent(epoll_event);
+
+impl Deref for EpollEvent {
+    type Target = epoll_event;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for EpollEvent {
+    fn default() -> Self {
+        EpollEvent(epoll_event {
+            events: 0u32,
+            u64: 0u64,
+        })
+    }
+}
+
+impl EpollEvent {
+    /// Create a new epoll_event instance with the following fields: `events`, which contains
+    /// an event mask and `data` which represents a user data variable. `data` field can be
+    /// a fd on which we want to monitor the events specified by `events`.
+    pub fn new(events: EventSet, data: u64) -> Self {
+        EpollEvent(epoll_event {
+            events: events.bits(),
+            u64: data,
+        })
+    }
+
+    /// Returns the `events` from `libc::epoll_event`.
```

Comment:
Wherever there's a link-able reference to another crate, please add the link so that the generated documentation is easily navigable. See https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html.

Reply:
:+1: